### ISSUE TITLE
question about what dataset to download

### ISSUE THREAD

**Issue by RandyChen233** [EduardoTayupanta/VisualOdometry#7]
Hi,

I wonder if you could clarify which dataset should we download ? I followed [this repository](https://github.com/ChiWeiHsiao/DeepVO-pytorch) to download all the visual odometry video frames . However, when I set the dataset path to the directory of images I've downloaded, I get the following error suggesting that multiple images are not found:

```
...
2023-05-14 06:40:41.661937: I tensorflow/core/common_runtime/executor.cc:1197] [/job:localhost/replica:0/task:0/device:CPU:0] (DEBUG INFO) Executor start aborting (this does not indicate an error and you can ignore this message): NOT_FOUND: /mnt/d/DeepVO-pytorch/KITTI/images/10/      1061.png; No such file or directory
         [[{{node ReadFile_1}}]]
  File "/home/randychen233/anaconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/ops/gen_dataset_ops.py", line 3016, in iterator_get_next
2023-05-14 06:40:41.662287: I tensorflow/core/common_runtime/executor.cc:1197] [/job:localhost/replica:0/task:0/device:CPU:0] (DEBUG INFO) Executor start aborting (this does not indicate an error and you can ignore this message): NOT_FOUND: /mnt/d/DeepVO-pytorch/KITTI/images/00/       520.png; No such file or directory
         [[{{node ReadFile_1}}]]
2023-05-14 06:40:41.662705: W tensorflow/core/framework/op_kernel.cc:1830] OP_REQUIRES failed at whole_file_read_ops.cc:114 : NOT_FOUND: /mnt/d/DeepVO-pytorch/KITTI/images/00/      1330.png; No such file or directory
2023-05-14 06:40:41.662993: I tensorflow/core/common_runtime/executor.cc:1197] [/job:localhost/replica:0/task:0/device:CPU:0] (DEBUG INFO) Executor start aborting (this does not indicate an error and you can ignore this message): NOT_FOUND: /mnt/d/DeepVO-pytorch/KITTI/images/02/      1653.png; No such file or directory
         [[{{node ReadFile_1}}]]
    _ops.raise_from_not_ok_status(e, name)
2023-05-14 06:40:41.664281: I tensorflow/core/common_runtime/executor.cc:1197] [/job:localhost/replica:0/task:0/device:CPU:0] (DEBUG INFO) Executor start aborting (this does not indicate an error and you can ignore this message): NOT_FOUND: /mnt/d/DeepVO-pytorch/KITTI/images/02/      3364.png; No such file or directory
         [[{{node ReadFile_1}}]]
  File "/home/randychen233/anaconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 7262, in raise_from_not_ok_status
...
```
Moreover, the dataset I have uses a 10-digit index for each `.png` file, so I changed the original `%04d` formatter to `%10d`, but it won't work

**Comment by ggonzalez2022** [comment]
KITTI Dataset works for me, I downloaded it manually (RGB dataset)

BTW do you have any idea of how to change the convert_ckp ...  work with FlowNetC and not FlowNetS ?

import os
import tensorflow as tf
from tensorflow import keras
from keras import layers


class FlowNetS:
    def __init__(self, height, width):
        conv1 = layers.Conv2D(64, kernel_size=7, strides=2, padding='same', input_shape=(height, width, 6),
                              name='conv1')
        leaky_relu_1 = layers.LeakyReLU(0.1)

        conv2 = layers.Conv2D(128, kernel_size=5, strides=2, padding='same', name='conv2')
        leaky_relu_2 = layers.LeakyReLU(0.1)

        conv3 = layers.Conv2D(256, kernel_size=5, strides=2, padding='same', name='conv3')
        leaky_relu_3 = layers.LeakyReLU(0.1)

        conv3_1 = layers.Conv2D(256, kernel_size=3, strides=1, padding='same', name='conv3_1')
        leaky_relu_3_1 = layers.LeakyReLU(0.1)

        conv4 = layers.Conv2D(512, kernel_size=3, strides=2, padding='same', name='conv4')
        leaky_relu_4 = layers.LeakyReLU(0.1)

        conv4_1 = layers.Conv2D(512, kernel_size=3, strides=1, padding='same', name='conv4_1')
        leaky_relu_4_1 = layers.LeakyReLU(0.1)

        conv5 = layers.Conv2D(512, kernel_size=3, strides=2, padding='same', name='conv5')
        leaky_relu_5 = layers.LeakyReLU(0.1)

        conv5_1 = layers.Conv2D(512, kernel_size=3, strides=1, padding='same', name='conv5_1')
        leaky_relu_5_1 = layers.LeakyReLU(0.1)

        conv6 = layers.Conv2D(1024, kernel_size=3, strides=2, padding='same', name='conv6')
        leaky_relu_6 = layers.LeakyReLU(0.1)

        self.model = keras.Sequential([
            conv1,
            leaky_relu_1,
            conv2,
            leaky_relu_2,
            conv3,
            leaky_relu_3,
            conv3_1,
            leaky_relu_3_1,
            conv4,
            leaky_relu_4,
            conv4_1,
            leaky_relu_4_1,
            conv5,
            leaky_relu_5,
            conv5_1,
            leaky_relu_5_1,
            conv6,
            leaky_relu_6,
        ])


path = '/media/German/VO/FlowNet/check/FlowNetS'
file = 'flownet-S.ckpt-0'
trained_checkpoint_prefix = os.path.join(path, file)
net = FlowNetS(375, 1242)

names = [
    'conv1',
    'conv2',
    'conv3',
    'conv3_1',
    'conv4',
    'conv4_1',
    'conv5',
    'conv5_1',
    'conv6'
]

weights = {}
biases = {}
for name in names:
    weights['FlowNetS/' + name + '/weights:0'] = None
    biases['FlowNetS/' + name + '/biases:0'] = None

loaded_graph = tf.Graph()
with tf.compat.v1.Session(graph=loaded_graph) as sess:
    loader = tf.compat.v1.train.import_meta_graph(trained_checkpoint_prefix + '.meta')
    loader.restore(sess, trained_checkpoint_prefix)

    variables_names = [v.name for v in tf.compat.v1.trainable_variables()]
    values = sess.run(variables_names)
    for k, v in zip(variables_names, values):
        if k == 'FlowNetS/conv1/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv2/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv3/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv3_1/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv4/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv4_1/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv5/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv5_1/weights:0':
            weights[k] = v
        elif k == 'FlowNetS/conv6/weights:0':
            weights[k] = v

        if k == 'FlowNetS/conv1/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv2/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv3/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv3_1/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv4/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv4_1/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv5/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv5_1/biases:0':
            biases[k] = v
        elif k == 'FlowNetS/conv6/biases:0':
            biases[k] = v

i = 0
for name_weights, name_biases in zip(weights, biases):
    l = []
    l.append(weights[name_weights])
    l.append(biases[name_biases])
    net.model.layers[i].set_weights(l)
    i += 2

net.model.save('flow.h5')

**Comment by ggonzalez2022** [comment]
KITTI Dataset works for me, I downloaded it manually (RGB dataset)

BTW do you have any idea of how to change the convert_ckp ... work with FlowNetC and not FlowNetS ?

`import os
import tensorflow as tf
from tensorflow import keras
from keras import layers

class FlowNetS:
def init(self, height, width):
conv1 = layers.Conv2D(64, kernel_size=7, strides=2, padding='same', input_shape=(height, width, 6),
name='conv1')
leaky_relu_1 = layers.LeakyReLU(0.1)

    conv2 = layers.Conv2D(128, kernel_size=5, strides=2, padding='same', name='conv2')
    leaky_relu_2 = layers.LeakyReLU(0.1)

    conv3 = layers.Conv2D(256, kernel_size=5, strides=2, padding='same', name='conv3')
    leaky_relu_3 = layers.LeakyReLU(0.1)

    conv3_1 = layers.Conv2D(256, kernel_size=3, strides=1, padding='same', name='conv3_1')
    leaky_relu_3_1 = layers.LeakyReLU(0.1)

    conv4 = layers.Conv2D(512, kernel_size=3, strides=2, padding='same', name='conv4')
    leaky_relu_4 = layers.LeakyReLU(0.1)

    conv4_1 = layers.Conv2D(512, kernel_size=3, strides=1, padding='same', name='conv4_1')
    leaky_relu_4_1 = layers.LeakyReLU(0.1)

    conv5 = layers.Conv2D(512, kernel_size=3, strides=2, padding='same', name='conv5')
    leaky_relu_5 = layers.LeakyReLU(0.1)

    conv5_1 = layers.Conv2D(512, kernel_size=3, strides=1, padding='same', name='conv5_1')
    leaky_relu_5_1 = layers.LeakyReLU(0.1)

    conv6 = layers.Conv2D(1024, kernel_size=3, strides=2, padding='same', name='conv6')
    leaky_relu_6 = layers.LeakyReLU(0.1)

    self.model = keras.Sequential([
        conv1,
        leaky_relu_1,
        conv2,
        leaky_relu_2,
        conv3,
        leaky_relu_3,
        conv3_1,
        leaky_relu_3_1,
        conv4,
        leaky_relu_4,
        conv4_1,
        leaky_relu_4_1,
        conv5,
        leaky_relu_5,
        conv5_1,
        leaky_relu_5_1,
        conv6,
        leaky_relu_6,
    ])
path = '/media/German/VO/FlowNet/check/FlowNetS'
file = 'flownet-S.ckpt-0'
trained_checkpoint_prefix = os.path.join(path, file)
net = FlowNetS(375, 1242)

names = [
'conv1',
'conv2',
'conv3',
'conv3_1',
'conv4',
'conv4_1',
'conv5',
'conv5_1',
'conv6'
]

weights = {}
biases = {}
for name in names:
weights['FlowNetS/' + name + '/weights:0'] = None
biases['FlowNetS/' + name + '/biases:0'] = None

loaded_graph = tf.Graph()
with tf.compat.v1.Session(graph=loaded_graph) as sess:
loader = tf.compat.v1.train.import_meta_graph(trained_checkpoint_prefix + '.meta')
loader.restore(sess, trained_checkpoint_prefix)

variables_names = [v.name for v in tf.compat.v1.trainable_variables()]
values = sess.run(variables_names)
for k, v in zip(variables_names, values):
    if k == 'FlowNetS/conv1/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv2/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv3/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv3_1/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv4/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv4_1/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv5/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv5_1/weights:0':
        weights[k] = v
    elif k == 'FlowNetS/conv6/weights:0':
        weights[k] = v

    if k == 'FlowNetS/conv1/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv2/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv3/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv3_1/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv4/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv4_1/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv5/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv5_1/biases:0':
        biases[k] = v
    elif k == 'FlowNetS/conv6/biases:0':
        biases[k] = v
i = 0
for name_weights, name_biases in zip(weights, biases):
l = []
l.append(weights[name_weights])
l.append(biases[name_biases])
net.model.layers[i].set_weights(l)
i += 2

net.model.save('flow.h5')
`

**Comment by EduardoTayupanta** [comment]
Hi, 
You can use this repository
[https://www.cvlibs.net/datasets/kitti/eval_odometry.php]
